File: codes/2024.12/1.js
// 题目: 给定一个无序的整数数组，找到其中第 k 大的元素。
// 输入: [3,2,1,5,6,4], k = 2
// 输出: 5
//
// 输入: [3,2,3,1,2,4,5,5,6], k = 4
// 输出: 4

function fn(nums, k) {

    nums.sort((a, b) => {
        return b - a;
    })

    return nums[k - 1]
}


fn([3, 2, 3, 1, 2, 4, 5, 5, 6], 4)
fn([3, 2, 1, 5, 6, 4], 2)

File: codes/2024/2024.09/5/1.py
# 优先级队列：基于有序列表的实现

class PriorityQueue:
    def __init__(self):
        # 用于存储元素的列表
        self.queue = []

    # 判断队列是否为空
    def isEmpty(self):
        return len(self.queue) == 0

    # 插入元素,需要传入两个参数，插入以后按照优先级排序
    #   item: 待插入的元素
    #   priority: 优先级
    def push(self, item, priority):
        self.queue.append((item, priority))
        self.queue.sort(key=lambda x: x[1])

    # 移除元素，返回优先级最高的元素，即返回列表的第一个元素
    def pop(self):
        # 参数 0 表示移除并返回列表的第一个元素
        return self.queue.pop(0)
